### DISCUSSION TITLE
Data collection without the online portal

### DISCUSSION THREAD

**Discussion by andreabdev** [universam1/iSpindel#537] [General]
Is it possible to collect data using a local wifi network without having an account for the online portal?
For example with a local web server or MQTT server or other API which allow to develop a custom software for data collection?
Thanks.

**Comment by Andlix** [comment]
Sure.
https://www.ispindel.de/docs/README_en.html#mqtt

**Comment by pppedrillo** [comment]
Yes, it is possible in many ways.

**Comment by mgbasi** [comment]
Yes. While I was running some tests, I set up a server in a raspberry pi. The iSpindle connects to the pi, send it the data, which the pi then stores in a JSON file. Unfortunately, you do need to know how to set up a server and application on a Pi for this to work.

![042F6034-F88C-4FEC-9DD6-655C9CAB721C](https://user-images.githubusercontent.com/69357692/134953197-148fc056-5b0f-43e1-a3db-afc6893b915f.png)

**Reply by pppedrillo** [reply]
> Unfortunately, you do need to know how to set up a server and application on a Pi for this to work.

Its fairly easy. To make this server work, you simply can follow it's "readme/how to" guide.

**Reply by mgbasi** [reply]
The iSpindel part is easy. The more difficult part is the RPi server that receives and parses the data, and displays it in a useful way. I mean, it’s not super difficult but you need the skills to do it.

**Reply by blueharmony** [reply]
I'm a software developer and familiar with Raspberry Pi hardware. Can you please detail what you did to setup the RPi server?

**Reply by mgbasi** [reply]
Yeah, I’ll clean up my code this week and put it in a public repository.

Basically, it is a NodeJs, Express, EJS server. The iSpindel posts data to a url at the port the Node application is watching, and then the app/server adds it to a JSON data file. To view the data, the server pulls from the JSON file and creates a table for each device with a data set, putting the pertinent data into cells.

Anyway, I’ll clean up my code and write some basic documentation this week. I’ll post it here when I have it uploaded.

**Reply by pppedrillo** [reply]
@mgbasi @blueharmony  You may  have a look @this https://github.com/avollkopf/iSpindel-TCP-Server
